frontend/components/home.jsx

```diff
@@ -57,56 +74,246 @@ class Home extends React.Component {
+          nextTrackURL: this.state.trackMetaData[this.getDate(this.state.charts, this.i + 1)]['previewUrl'],
+          trackURLSoundComponentOne: this.state.trackMetaData[this.getDate(this.state.charts, this.i + 1)]['previewUrl'],
+          trackURLSoundComponentTwo: this.state.trackMetaData[this.getDate(this.state.charts, this.i)]['previewUrl'],
+          volOne: 0,
+          volTwo: 100
+        });
+        if(this.state.isSoundOn) {
+          this.setState({
+              soundComponentOneStatus: Sound.status.STOPPED,
+              soundComponentTwoStatus: Sound.status.PLAYING
+          });
+        }
+    } else {
+      this.activeSoundComponent = "one";
+      this.setState({
+          currentDate: this.getDate(this.state.charts, this.i),
+          nextChartDate: this.getDate(this.state.charts, this.i + 1),
+          currentTrackURL: this.state.trackMetaData[this.getDate(this.state.charts, this.i)]['previewUrl'],
+          nextTrackURL: this.state.trackMetaData[this.getDate(this.state.charts, this.i + 1)]['previewUrl'],
+          trackURLSoundComponentOne: this.state.trackMetaData[this.getDate(this.state.charts, this.i)]['previewUrl'],
+          trackURLSoundComponentTwo: this.state.trackMetaData[this.getDate(this.state.charts, this.i + 1)]['previewUrl'],
+          volOne: 100,
+          volTwo: 0
+        });
+      if(this.state.isSoundOn) {
+        this.setState({
+          soundComponentOneStatus: Sound.status.PLAYING,
+          soundComponentTwoStatus: Sound.status.STOPPED
+        });
+      }
+    }
+
     this.i += 1;
+    this.fadeInFadeOut();
     this.createInterval();
   }
 
   getDate(charts, index) {
     return Object.keys(charts)[index];
   }
 
-  toggleSound() {
-    this.setState({ soundPlaying: !this.state.soundPlaying });
+  formatDate(date) {
+    return moment(date).format('MMMM D, YYYY');
   }
 
-  handleSongFinishedPlaying() {
-    // resets the states currentTrackURL value if the song sample is finished
-    this.setState({ currentTrackURL: this.state.currentTrackURL });
+  isNextSongDifferent(){
+    return this.state.currentTrackURL !== this.state.nextTrackURL;
   }
 
-  formatDate(date) {
-    return moment(date).format('MMMM D, YYYY');
+  areBothPlaying() {
+    return (this.state.soundComponentTwoStatus === Sound.status.PLAYING) && (this.state.soundComponentOneStatus === Sound.status.PLAYING);
+  }
+
+  fadeInFadeOut() {
+    if (this.isNextSongDifferent()) {
+      if (this.activeSoundComponent === 'one') {
+        if (this.fadeOutOneInTwo) {
+          clearInterval(this.fadeOutOneInTwo);
+        }
+        if (this.fadeOutTwoInOne) {
+          clearInterval(this.fadeOutTwoInOne);
+        }
+        this.fadeOutOneInTwo = setInterval(() => {
+          this.setState({
+            volOne: this.state.volOne - 1.5,
+            volTwo: this.state.volTwo + 1.5
+          });
+        }, (1000 / 30));
+      } else {
+        if (this.fadeOutTwoInOne) {
```

Comment:
Just go ahead and clear interval at top of func since you're going to be doing it anyway